### PR TITLE
provider/aws: convert aws rds subnet to aws-sdk-go

### PR DESCRIPTION
**Pull request by catsby**
This completes the migration for AWS RDS to aws-sdk-go. 

If #1064 gets merged, I can update this to include changes to `config/aws.go` to completely replace `goamz/rds`

### CODE REVIEW

File: builtin/providers/aws/resource_aws_db_subnet_group.go [hunk 1]
```diff
@@ -82,12 +83,17 @@ func resourceAwsDbSubnetGroupRead(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	if len(describeResp.DBSubnetGroups) != 1 ||
```
Comment (pearkes) [inline builtin/providers/aws/resource_aws_db_subnet_group.go]: Looks like this conditional is doing nothing? Should we maybe return an error saying "More than one group returned"?

Also, for readability assigning the 0-indexed group to a var might be nice after this check.

Comment (catsby) [inline builtin/providers/aws/resource_aws_db_subnet_group.go]: I'm reading it as "if there is 0 or more than 1 :: OR :: the first one is not `d.Id()`". Seem right?

> Also, for readability assigning the 0-indexed group to a var might be nice after this check.

Legit

Comment (pearkes) [inline builtin/providers/aws/resource_aws_db_subnet_group.go]: Yea, that's right. It just needs to do something if those checks don't pass.

Comment (catsby) [inline builtin/providers/aws/resource_aws_db_subnet_group.go]: Got it – refactored in  db37b34 to fix this, take another look when you get a chance

**Comment by phinze** [comment]
Mind rebasing to clean up these commits?

provider_aws__convert_aws_rds_subnet_to_aws-sdk-go_by_catsby_ _pull_request__1067_ _hashicorp_terraform

Unless you are a branch history purist, in which case I will defer to your religious preferences. :bowtie:

**Comment by phinze** [comment]
Other than my git nit pick, LGTM! :fist: :rocket:

**Comment by mitchellh** [comment]
LGTM here too.

**Pull request merged**